sqlglot/dialects/duckdb.py

```diff
@@ -68,6 +68,15 @@ def _struct_pack_sql(self, expression):
     return f"STRUCT_PACK({', '.join(args)})"
 
 
+def _datatype_sql(self, expression):
+    type_value = expression.this
+    if type_value == exp.DataType.Type.ARRAY:
+        interior = self.expressions(expression, flat=True)
```

Comment:
sorry to nit but i'm not a fan of this style can we just do

if expression.this == exp.DataType.Type.ARRAY:
    return f"{self.expressions(expression, flat=True)}[]"
return self.datatype_sql(expression)